bin/build.js

```diff
@@ -179,41 +183,45 @@ function cleanup() {
 }
 
 function buildPluginsForBrowserify() {
-  return Promise.all(plugins.map(function (plugin) {
-    return doRollup('src_browser/plugins/' + plugin + '/index.js',
-      'lib/extras/' + plugin + '.js');
-  }));
+  return mkdirp('lib/extras').then(function() {
+    return Promise.all(plugins.map(function (plugin) {
+      return doRollup('src_browser/plugins/' + plugin + '/index.js',
+                      'lib/extras/' + plugin + '.js');
+    }));
+  });
 }
 
 function buildExtras() {
-  return Promise.all(Object.keys(extras).map(function (entry) {
-    var target = extras[entry];
-    return doRollup('src_browser/' + entry, 'lib/extras/' + target);
-  }));
+  return mkdirp('lib/extras').then(function() {
+    return Promise.all(Object.keys(extras).map(function (entry) {
+      var target = extras[entry];
+      return doRollup('src_browser/' + entry, 'lib/extras/' + target);
+    }));
+  });
 }
 
 function buildPluginsForBrowser() {
-  return Promise.all(plugins.map(function (plugin) {
-    var source = 'lib/extras/' + plugin + '.js';
-    return doBrowserify(source, {}, 'pouchdb').then(function (code) {
-      code = comments[plugin] + code;
-      return Promise.all([
-        writeFile('dist/pouchdb.' + plugin + '.js', code),
-        doUglify(code, comments[plugin], 'dist/pouchdb.' + plugin + '.min.js')
-      ]);
-    });
-  }));
+  return mkdirp('lib/extras').then(function() {
+    return Promise.all(plugins.map(function (plugin) {
+      var source = 'lib/extras/' + plugin + '.js';
+      return doBrowserify(source, {}, 'pouchdb').then(function (code) {
+        code = comments[plugin] + code;
+        return Promise.all([
+          writeFile('dist/pouchdb.' + plugin + '.js', code),
+          doUglify(code, comments[plugin], 'dist/pouchdb.' + plugin + '.min.js')
+        ]);
+      });
+    }));
+  });
+}
+
+if (process.argv[2] === 'node') {
+  buildForNode();
+  return;
 }
 
-Promise.resolve().then(function () {
-  return Promise.all(['dist', 'lib', 'src_browser'].map(function (file) {
-    return rimraf(file);
```

Comment:
Removing `rimraf()` here means that it's possible to accidentally commit unwanted files from the `dist ` directory